plasma_framework/contracts/src/exits/payment/controllers/PaymentChallengeIFENotCanonical.sol

```diff
@@ -85,6 +91,35 @@ library PaymentChallengeIFENotCanonical {
         emit InFlightExitChallenged(msg.sender, keccak256(args.inFlightTx), competitorPosition);
     }
 
+    function respond(
+        Controller memory self,
+        PaymentExitDataModel.InFlightExitMap storage inFlightExitMap,
+        bytes memory inFlightTx,
+        uint256 inFlightTxPos,
+        bytes memory inFlightTxInclusionProof
+    )
+        public
+    {
+        uint192 exitId = ExitId.getInFlightExitId(inFlightTx);
+        PaymentExitDataModel.InFlightExit storage ife = inFlightExitMap.exits[exitId];
+        require(ife.exitStartTimestamp != 0, "In-flight exit doesn't exists");
+
+        require(
+            ife.oldestCompetitorPosition > inFlightTxPos,
+            "In-flight transaction has to be younger than competitors to respond to non-canonical challenge.");
+
+        UtxoPosLib.UtxoPos memory utxoPos = UtxoPosLib.UtxoPos(inFlightTxPos);
```

Comment:
this line looks weird, why casting a `txPos` to become `utxoPos`.

And I just realized in `RootChain.sol` it is saving `txPos` to the `competitorPosition` and not `utxoPos`. I think we should fix that for the `run` function too. (Also, I'd like to recommend to rename `run` to `challenge` if you'd like to put the two functions in same file)

Reply:
Aaaa :man_facepalming: you're absolutely right :exclamation: Nice catch :exclamation: 
 :+1: rename `run -> challenge`

Reply:
Eeee, no :man_facepalming: it's totally misleading @boolafish  - client [provides UtxoPos to the contract](https://github.com/omisego/elixir-omg/blob/6df52caa9f7aee079cb9b7f78e63ee8d1d44a80d/apps/omg_watcher/lib/omg_watcher/exit_processor/canonicity.ex#L147) but `RootChain.sol` tricks you it's `txPos`. Oh my ... :man_facepalming:

Reply:
hmmm....but for the case of competitor position, shouldn't it be considering the position of the tx (thus txPos)? because you're competing the priority as a tx not as an output, isn't it?

Reply:
🤦‍♂.....but yeah, it seems like RootChain.sol is using variable name `txPos` but being a `utxoPos` @@

Reply:
but then you can challenge or respond multiple time with different outputIndexs if we're using utxoPos. As the lower outputIndex can override the one with higher outputIndex number during challenge and respond. Probably nothing would be harmed? However, it seems to me txPos is more logically correct?

Reply:
Luckily in canonicity game we are interested in oldest competitor so the one with lowest `TxoPos`
so if one challenge with `(blknum, txindex, oindex) = (1000, 222, 999)` you can take his bond easily providing the same data but `TxoPos = (1000, 222, 0)`  which corresponds to the same Transaction position (where output index is ignored)

Reply:
yeah, I agree with that so probably nothing would be harmed. However, this means our client should know that, hey, even we are asking a utxoPos, but please just always give as a txPos + 0, otherwise you might not get your bond......

I don't see this as better design as why we are forcing our client to remember always passing a 0 that should never be used? To me it seems like `RootChain.sol` have the right interface and var naming but implemented wrong instead of the opposite though.

Reply:
Anyway - I create #254 to avoid/limit spreading the confusion. This is HUGE security thread if we mess with this and use both interchangeably...

Reply:
ah...I know this is merge but I think we should rename the `inFlightTxPos` to `inFlightUTxoPos` to not introduce more confusion. Whether txPos or utxoPos we can leave it to the issue #254.

Reply:
Yeah, there are more them as `competitorTxPos` and so. I'm leaving it for #254.